torch/nn/modules/activation.py

```diff
@@ -756,7 +762,7 @@ def forward(self, query, key, value, key_padding_mask=None,
         - attn_output_weights: :math:`(N, L, S)` where N is the batch size,
           L is the target sequence length, S is the source sequence length.
         """
-        if hasattr(self, '_qkv_same_embed_dim') and self._qkv_same_embed_dim is False:
+        if not self._qkv_same_embed_dim:
```

Comment:
do you mean `if self._qkv_same_embed_dim is None`?
`if not` can be ambiguous, if it's an integer of value `0`, or `False` or `None`